sdk/communication/azure-communication-identity/src/main/java/com/azure/communication/identity/CommunicationIdentityAsyncClient.java

```diff
@@ -113,50 +115,87 @@ public Mono<Response<CommunicationUserIdentifier>> createUserWithResponse() {
+     * @param tokenExpiresAfter Custom validity period of the Communication Identity access token within [1,24]
+     * hours range. If not provided, the default value of 24 hours will be used.
      * @return The created communication user and token.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<CommunicationUserIdentifierAndToken>
-        createUserAndToken(Iterable<CommunicationTokenScope> scopes) {
+        createUserAndToken(Iterable<CommunicationTokenScope> scopes, Duration tokenExpiresAfter) {
         try {
             Objects.requireNonNull(scopes);
             final List<CommunicationTokenScope> scopesInput = StreamSupport.stream(scopes.spliterator(), false).collect(Collectors.toList());
-            return client.createAsync(new CommunicationIdentityCreateRequest().setCreateTokenWithScopes(scopesInput))
+
+            CommunicationIdentityCreateRequest communicationIdentityCreateRequest =
+                utils.createCommunicationIdentityCreateRequest(scopesInput, tokenExpiresAfter, logger);
+
+            return client.createAsync(communicationIdentityCreateRequest)
                 .onErrorMap(CommunicationErrorResponseException.class, e -> translateException(e))
                 .flatMap(
                     (CommunicationIdentityAccessTokenResult result) -> {
                         return Mono.just(userWithAccessTokenResultConverter(result));
                     });
-        } catch (RuntimeException ex) {
+        }
+        catch (RuntimeException ex) {
             return monoError(logger, ex);
         }
     }
 
+    /**
+     * Creates a new CommunicationUserIdentifier with token.
+     *
+     * @param scopes The list of scopes for the token.
+     * @return The created communication user and token.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<CommunicationUserIdentifierAndToken>
+        createUserAndToken(Iterable<CommunicationTokenScope> scopes) {
+            return createUserAndToken(scopes, null);
+    }
+
     /**
      * Creates a new CommunicationUserIdentifier with token with response.
      *
      * @param scopes The list of scopes for the token.
+     * @param tokenExpiresAfter Custom validity period of the Communication Identity access token within [1,24]
+     * hours range. If not provided, the default value of 24 hours will be used.
      * @return The result with created communication user and token with response.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<Response<CommunicationUserIdentifierAndToken>>
-        createUserAndTokenWithResponse(Iterable<CommunicationTokenScope> scopes) {
+        createUserAndTokenWithResponse(Iterable<CommunicationTokenScope> scopes, Duration tokenExpiresAfter) {
         try {
             Objects.requireNonNull(scopes);
             final List<CommunicationTokenScope> scopesInput = StreamSupport.stream(scopes.spliterator(), false).collect(Collectors.toList());
```

Comment:
should be in the `createCommunicationIdentityCreateRequest` method, same for other occurrences and the sync client class

Reply:
Moved to corresponding methods.